src/lib/comms/sol-oic-security.c

```diff
@@ -266,11 +269,26 @@ creds_init(const void *data)
     creds->security = data;
     sol_vector_init(&creds->items, sizeof(struct cred_item));
 
-    file_data = sol_util_load_file_string(file_name, &length);
-    if (!file_data)
+    cert = sol_cert_load_from_file(file_name);
+    if (!cert)
         return creds;
 
-    sol_json_scanner_init(&scanner, file_data, length);
+    cert_size = sol_cert_size(cert);
+    SOL_INT_CHECK_GOTO(cert_size, < 0, error);
+    if (cert_size == 0)
+        goto out_cert;
+
+    file_data = malloc((size_t)cert_size);
+    SOL_NULL_CHECK_GOTO(file_data, error);
+
+    sol_buffer_init_flags(&buf, file_data, (size_t)cert_size,
+        SOL_BUFFER_FLAGS_CLEAR_MEMORY | SOL_BUFFER_FLAGS_NO_NUL_BYTE |
+        SOL_BUFFER_FLAGS_FIXED_CAPACITY);
+
+    r = sol_cert_read_data(cert, &buf);
+    SOL_INT_CHECK_GOTO(r, < 0, error_buf);
```

Comment:
this is not the best approach, sync with @ceolin as he did a nicer implementation.